src/macros.hpp

```diff
@@ -112,3 +112,32 @@
     namespace Aquamarine {                                                                                                                                                         \
         class name;                                                                                                                                                                \
     }
+
+#define EXTRACT_CLASS_NAME()                                                                                                                                                       \
```

Comment:
all this should be in IWaylandProtocol.hpp prolly

Reply:
moved it there